### PR TITLE
Add Southbound HTTPS support + Standardize MQTT

### PR DESCRIPTION
**Pull request by jason-fox**
- Allow HTTP endpoint to be configured with a cert and a private key
- Allow self signed certs when using MQTT with SSL
- Update documentation

### CODE REVIEW

File: docs/installationguide.md [hunk 1]
```diff
@@ -165,16 +165,30 @@ JavaScript file, that contains the following sections:
...
 -   **username**: Username for the IoT Agent in the MQTT broker, if authentication is activated.
 -   **password**: Password for the IoT Agent in the MQTT broker, if authentication is activated.
+-   **ca**: ca certificates to use for validating server certificates (optional). Default is to trust the well-known CAs
+    curated by Mozilla. Mozilla's CAs are completely replaced when CAs are explicitly specified using this option.
+-   **cert**: cert chains in PEM format to use for authenticating into the MQTT broker (optional). Only used when using
+    `mqtts`, `tls` or `wss` as connnection protocol.
+-   **key**: optional private keys in PEM format to use on the client-side for connecting with the MQTT broker
+    (optional). Only used when using `mqtts`, `tls` or `wss` as connection protocol. The included CA list will be used
+    to determine if server is authorized.
+-   **protocol**: MQTT protocol to use (default `mqtt`)
```
Comment (fgalan) [inline docs/installationguide.md:180]: Protocol appears twice.

File: package.json [hunk 2]
```diff
@@ -42,6 +42,7 @@
     "logops": "2.1.0",
     "mqtt": "3.0.0",
     "request": "2.88.0",
+    "sinon": "^9.0.1",
```
Comment (fgalan) [inline package.json:45]:
```suggestion
    "sinon": "~9.0.1",
```

Why this new depedency is needed? If related with testing it should be moved to `devDependencies`. I understand that sinon is something related with utest tooling... Cannot the current ones (mocha and nock) cover the same functionality?

File: config.js [hunk 3]
```diff
@@ -36,6 +36,12 @@ config.mqtt = {
      */
```
Comment (fgalan) [inline config.js:36]: CHANGES_NEXT_RELEASE entry about the support to HTTPS should be added.

In addition, I understand this PR provides MQTTS support so this issue would be closed by the PR https://github.com/telefonicaid/iotagent-ul/issues/354, wouldn't it? In that case, an entry about that should be also added about it.

Comment (jason-fox) [reply config.js:36]: Fixed 146025b

Comment (jason-fox) [reply config.js:36]: CNR updated to refer to  #354 - Fixed eb71a30

File: docs/installationguide.md (hunk 1 above)
Comment (jason-fox) [reply docs/installationguide.md]: Duplicate removed: eb71a30

File: package.json (hunk 2 above)
Comment (jason-fox) [reply package.json]: Fixed 240d5f9 Moved to `devDependencies` - it is required for mocking the file system.  Nock is used for HTTP mocks. Sinon was suggested via the previously commented code in the relevant test in the JSON Agent.

:bar_chart: Got to keep that code coverage up. :chart_with_upwards_trend: 84% and rising

File: docs/installationguide.md [hunk 4]
```diff
@@ -165,16 +165,29 @@ JavaScript file, that contains the following sections:
 The `config.mqtt` section of the config file contains all the information needed to connect to the MQTT Broker from the
 IoT Agent. The following attributes are accepted:
 
+-   **protocol**: protocol to use for connecting with the MQTT broker (`mqtt`, `mqtts`, `tcp`, `tls`, `ws`, `wss`).
```
Comment (fgalan) [inline docs/installationguide.md:168]: It would be usefault to explain which one is the default (I understand that `mqtt`, for backward compatibility).

Comment (jason-fox) [reply docs/installationguide.md:168]: Fixed a4adf6e

**Pull request merged**